dmenufm

```diff
@@ -21,159 +21,213 @@
-[ -n "$FM_NOTIF_FONT" ] || FM_NOTIF_FONT="Monospace-25"
-[ -n "$FM_DANGER_FONT" ] || FM_DANGER_FONT="Monospace-20"
+[ -n "$FM_GENERIC_FONT" ] || FM_GENERIC_FONT='Monospace-15'
+[ -n "$FM_NOTIF_FONT" ] || FM_NOTIF_FONT='Monospace-25'
+[ -n "$FM_DANGER_FONT" ] || FM_DANGER_FONT='Monospace-20'
 # COLORs
-[ -n "$FM_GENERIC_COLOR" ] || FM_GENERIC_COLOR="#005577"
-[ -n "$FM_ACTION_COLOR_LV1" ] || FM_ACTION_COLOR_LV1="#33691e"
-[ -n "$FM_ACTION_COLOR_LV2" ] || FM_ACTION_COLOR_LV2="#FF8C00"
-[ -n "$FM_ACTION_COLOR_BULK" ] || FM_ACTION_COLOR_BULK="#CB06CB"
+[ -n "$FM_GENERIC_COLOR" ] || FM_GENERIC_COLOR='#005577'
+[ -n "$FM_ACTION_COLOR_LV1" ] || FM_ACTION_COLOR_LV1='#33691e'
+[ -n "$FM_ACTION_COLOR_LV2" ] || FM_ACTION_COLOR_LV2='#FF8C00'
+[ -n "$FM_ACTION_COLOR_BULK" ] || FM_ACTION_COLOR_BULK='#CB06CB'
 
 # xdg-open directories
-XDGDIR1="/usr/share/applications"
-XDGDIR2="/usr/local/share/applications"
+XDGDIR1='/usr/share/applications'
+XDGDIR2='/usr/local/share/applications'
 
 ## GLOBAL VARIABLES
-CHOICE="placeholder"
-actCHOICE="placeholder"
-allselection="placeholder"
-TARGET="./"
-BACKWARD="../"
-ENDSELECTION="End Selection"
-allowbulk="NotAllowed"
-ACTION="Actions"
-TERM="Terminal"
-FM_PCP="PCP - Copy path"
-FM_NEW="NEW - Create new file / directory"
-FM_DEL="DEL - Delete files / directories"
-FM_MVR="MVR - Move files / directories"
-FM_YAK="YAK - Copy files / directories"
-FM_LNK="LNK - Symbolically link files / directories"
-FM_REM="REM - Rename files / directories"
-FM_TRH="TRH - Trash of dmenufm"
-FM_HIS="HIS - History of dmenufm"
-FM_BMK="BMK - Bookmark for dmenufm"
-FM_CMD="CMD - Frequently used command"
-FM_ZIP="ZIP - Compress files / directories"
+CHOICE='placeholder'
+actCHOICE='placeholder'
+allselection='placeholder'
+TARGET='./'
+BACKWARD='../'
+ENDSELECTION='End Selection'
+allowbulk='NotAllowed'
+ACTION='Actions'
+TERM='Terminal'
+FM_PCP='PCP - Copy path'
+FM_NEW='NEW - Create new file / directory'
+FM_DEL='DEL - Delete files / directories'
+FM_MVR='MVR - Move files / directories'
+FM_YAK='YAK - Copy files / directories'
+FM_LNK='LNK - Symbolically link files / directories'
+FM_REM='REM - Rename files / directories'
+FM_TRH='TRH - Trash of dmenufm'
+FM_HIS='HIS - History of dmenufm'
+FM_BMK='BMK - Bookmark for dmenufm'
+FM_CMD='CMD - Frequently used command'
+FM_ZIP='ZIP - Compress files / directories'
 ACTLIST="$FM_PCP:$FM_NEW:$FM_MVR:$FM_YAK:$FM_LNK:$FM_DEL:$FM_TRH:$FM_REM:$FM_HIS:$FM_BMK:$FM_CMD:$FM_ZIP"
-COMPRESSIONLIST="tar.gz:tar.bz2:tar.xz:xz:lzma:gz:bz2:7z:zip"
+COMPRESSIONLIST='tar.gz tar.bz2 tar.xz xz lzma gz bz2 7z zip'
 
 ### FUNCTIONS
 
 ## PRE-DETERMINATION FUNCTION
 
-# Check os
-get_os () {
-	case $OSTYPE in
-		# Mac OS X / macOS.
-		darwin*) FM_OPENER=open ;;
-	esac
+CheckDeps () {
+	DepCount=0
+	for CurDep in "$@"; do
+		if ! command -v "$CurDep" 1> /dev/null 2>&1; then
+			DepCount=$((DepCount + 1))
+			printf "ERROR: Dependency '$CurDep' not met." 1>&2
+		fi
+	done
+
+	[ $DepCount -eq 0 ] || exit 1
```

Comment:
If you exit 1, then what's the point of printing error? In my case, the whole terminal ended, and user cannot see the error message.

Reply:
For debugging purposes.